### ISSUE TITLE
Connection pool

### ISSUE THREAD

**Issue by aman28061994** [brettwooldridge/HikariCP#1025]
#1024  ### Environment
```
Hii brett i'm quite new with the dbPool thing , the thing is that i'm having connections in my DbPool available for eg i have pool size of 1000 and after acquiring around 500 connection the error connection timeout is thrown.

Can you please guide me on this.
HikariCP version: x.x.x
JDK version     : 1.8.0_111
Database        : PostgreSQL|MySQL|...
Driver version  : x.x.x
```
-----------------------------------------------------------------------------------------
*Have you searched the CLOSED issues already? How about checking [stackoverflow](https://stackoverflow.com/search?q=HikariCP)?*

**Comment by brettwooldridge** [comment]
@aman28061994 1000 connections is totally insane.  You are clearly reaching a DB limit, not a limit in HikariCP.

[Please read this page about pool sizing.](https://github.com/brettwooldridge/HikariCP/wiki/About-Pool-Sizing)  There is a video as well, from the Oracle Real World Performance series, about pool sizing in which they demonstrate that an application with a pool of 96 connections performs almost 50x faster than one with 1000 (or 4000) connections.

Even 96 connections is very large unless you have a 48-core server.

Basically, the formula is this:
```
connections = ((core_count * 2) + effective_spindle_count)
```
Where "effective_spindle_count" is the number of storage disks.  For a simple 4-core server with one hard disk this means 9 connection pool -- call it 10 as a round number.

**Comment by scopendo** [comment]
Hi @brettwooldridge – could you clarify a question on the connection pool sizing that I still have after reading the article and watching the video.  Is the size of the connection pool dependent on the number of cores of the application server(s) or the database server?  Obviously the DB server can only serve a given number of transactions at a time, but the connection pool settings are on the app server side as I understand, where there are N (possibly dynamic) app servers.

Thank you for the info so far – like many, I suspect our pool size is too high.

**Comment by billoneil** [comment]
See this https://github.com/brettwooldridge/HikariCP/issues/1023#issuecomment-345592654
